.github/workflows/ci.yaml

```diff
@@ -22,42 +25,233 @@ permissions:
   packages: write # to upload container
 
 jobs:
-
   metadata:
     runs-on: ubuntu-22.04
     outputs:
-      TRIAL_BRANCH: ${{ steps.meta.outputs.TRIAL_BRANCH }}
+      BUILD_DATE: ${{ steps.date.outputs.BUILD_DATE }}
+      PUBLISH: ${{ steps.if-publish.outputs.PUBLISH }}
     steps:
-      - name: Set optional trial branch
-        id: meta
+      - name: Set build date
+        id: date
+        shell: bash -x -e {0}
+        run: |
+          BUILD_DATE=$(TZ='US/Los_Angeles' date '+%Y-%m-%d')
+          echo "BUILD_DATE=${BUILD_DATE}" >> $GITHUB_OUTPUT
+
+      - name: Determine whether results will be 'published'
+        id: if-publish
         shell: bash -x -e {0}
         run: |
-          if [[ -z "${{ inputs. TRIAL_BRANCH}}" ]]; then
-            echo "TRIAL_BRANCH=${{ github.sha }}"
-          else
-            echo "TRIAL_BRANCH=${TRIAL_BRANCH}"
-          fi | tee $GITHUB_OUTPUT
+          echo "PUBLISH=${{ github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && inputs.PUBLISH) }}" >> $GITHUB_OUTPUT
 
   amd64:
     needs: metadata
     uses: ./.github/workflows/_ci.yaml
     with:
       ARCHITECTURE: amd64
-      TRIAL_BRANCH: ${{ needs.metadata.outputs.TRIAL_BRANCH }}
     secrets: inherit
  
   arm64:
     needs: metadata
     uses: ./.github/workflows/_ci.yaml
     with:
       ARCHITECTURE: arm64
-      TRIAL_BRANCH: ${{ needs.metadata.outputs.TRIAL_BRANCH }}
     secrets: inherit
 
+  publish-mealkit:
+    needs: [metadata, amd64, arm64]
+    if: (!cancelled()) && needs.metadata.outputs.PUBLISH
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - TAG: JAX
```

Comment:
Can this be simplified and derived from ALL_TAGS?

Reply:
I tried to prototype it and almost found the solution, but the problem is that I have 2 different `ALL_TAGS` (one from amd64 build, the other one is from arm64 one). I need to combine these 2 `ALL_TAGS` into 1 JSON object somehow  and feed `matrix` with it.
But let's create an issue for further pipeline improvements after I merge this PR.
Agree? @terrykong